scripts/us_census/acs5yr/subject_tables/common/generateColMap.py

```diff
@@ -0,0 +1,94 @@
+"""Module to generate a column map to their corresponding statistical variable defintions"""
+
+import os
+import json
+import pandas as pd
+
+#TODO:Remove this import and make it from ~/data/utils/ directory when it is available
+os.system('curl https://raw.githubusercontent.com/Abilityguy/data/dcid_gen/util/statvar_dcid_generator.py -LO statvar_dcid_generator.py')
+import statvar_dcid_generator
+
+dcid_gen = statvar_dcid_generator.StatVarDcidGenerator()
+
+class generateColMapBase:
+  """module to generate a column map given a list of columns of the dataset and a JSON Spec"""
+  def __init__(self, specDict=None, columnList=None):
+    """module init """
+    self.features = specDict
+    self.columnList = columnList
+    self.columnMap = {}
+    self.makeColMap()
+
+  def makeColMap(self):
+    #for each column generate the definition of their respective statistical variable node
+    for col in self.columnList:
+      if 'ignoreColumns' in self.features and col not in self.features['ignoreColumns']:
+          self.columnMap[col] = self.column_to_statVar(col)
+    #TODO: Enable validating the generate columnMap
+    #self.valid_colMap():
+    return self.columnMap
+
+  def column_to_statVar(self, column):
+    """generates a dictionary statistical variable with all properties specified in the JSON spec for a single column"""
+    base = False
+    statVar = {}
+    partList = column.split('!!')
```

Comment:
snake-case all variables?  https://google.github.io/styleguide/pyguide.html

Reply:
This will be resolved in the next commit.